biz.aQute.bndlib/src/aQute/bnd/cdi/AnnotationReader.java

```diff
@@ -0,0 +1,340 @@
+				Object[] serviceClasses = annotation.get("value");
+
+				if (serviceClasses != null && serviceClasses.length > 0) {
+					for (Object serviceClass : serviceClasses) {
+						definitions.get(0).service.add((TypeRef) serviceClass);
+					}
+				} else if (interfaces != null && interfaces.length > 0) {
+					for (TypeRef inter : interfaces) {
+						definitions.get(0).service.add(inter);
+					}
+				} else {
+					definitions.get(0).service.add(clazz.getClassName());
+				}
+				break;
+			default :
+				// this can't happen, @Service is limited to the above targets
+		}
+	}
+
+	@Override
+	public void classBegin(int access, TypeRef name) {
+		definitions.get(0).implementation = name;
+
+		PackageDef packageDef = packageInfos.computeIfAbsent(name.getPackageRef(), k -> {
+			Clazz packageInfoClazz = analyzer.getPackageInfo(k);
+			if (packageInfoClazz != null) {
+				try {
+					PackageDef pd = new PackageDef();
+					packageInfoClazz.parseClassFileWithCollector(pd);
+					return pd;
+				} catch (Exception e) {
+					analyzer.exception(e, "Error while processing package-info of class %s", clazz);
+				}
+			}
+			return new PackageDef();
+		});
+
+		if (packageDef.marked != null) {
+			definitions.get(0).marked = packageDef.marked.matches(name.getFQN());
+		}
+	}
+
+	@Override
+	public void classEnd() throws Exception {
+		member = null;
+		referenceDef = null;
+	}
+
+	@Override
+	public void extendsClass(TypeRef name) {
+		this.extendsClass = name;
+	}
+
+	@Override
+	public void field(FieldDef field) {
+		this.member = field;
+	}
+
+	@Override
+	public void implementsInterfaces(TypeRef[] interfaces) {
+		this.interfaces = interfaces;
+	}
+
+	@Override
+	public void memberEnd() {
+		member = null;
+		referenceDef = null;
+	}
+
+	@Override
+	public void method(MethodDef method) {
+		if (method.isAbstract() || method.isBridge())
+			return;
+
+		if (!baseclass && method.isPrivate())
+			return;
+
+		this.member = method;
+
+		if (member instanceof MethodDef) {
+			String signature = member.getSignature();
+			if (signature == null) {
+				signature = method.getDescriptor()
+					.toString();
+			}
+			MethodSignature sig = MethodSignature.of(signature);
+			if (sig.parameterTypes.length == 1) {
+				JavaTypeSignature javaTypeSignature = sig.parameterTypes[0];
+				// we only care about ClassTypeSignature
+				if (javaTypeSignature instanceof ClassTypeSignature) {
+					ClassTypeSignature parameterSig = (ClassTypeSignature) javaTypeSignature;
+					TypeRef typeRef = analyzer.getTypeRef(parameterSig.binary);
+					String fqn = typeRef.getFQN();
+					if (fqn.equals("org.osgi.service.cdi.reference.BindService")
+						|| fqn.equals("org.osgi.service.cdi.reference.BindBeanServiceObjects")
+						|| fqn.equals("org.osgi.service.cdi.reference.BindServiceReference")) {
+
+						TypeArgument typeArgument = parameterSig.classType.typeArguments[0];
+						ClassTypeSignature classSig = (ClassTypeSignature) typeArgument.type;
```

Comment:
Use a MethodResolver to get the type argument. See https://github.com/bndtools/bnd/blob/ceba0425550d6bd6bfb572e5419488cc042ddfd9/biz.aQute.bndlib/src/aQute/bnd/component/AnnotationReader.java#L1167-L1169

Also note, that type arguments may not be ClassTypeSignature to your cast can fail. You must test the type of the resolved type.